src/devices/ikea_sparsnas.c

```diff
@@ -0,0 +1,251 @@
+
+The IKEA Sparsnäs consists of a display unit, and a sender unit. The display unit 
+displays and stores the values sent by the sender unit. It is not needed for this
+decoder. The sender unit is placed by the energy meter. The sender unit has an
+IR photo sensor which is placed over the energy meter impulse diode. The sender
+also has an external antenna, which should be placed where it can provide non-
+interfered transmissions.
+
+The energy meter sends a fixed number of pulses per kWh. This is different per
+unit, but usual values are 500, 1000 and 2000. This is usually indicated like
+
+1000 imp/kWh
+
+on the front of the meter. This value goes into IKEA_SPARSNAS_PULSES_PER_KWH 
+in this file. The sender also has a unique ID which is used in the encryption
+key, hence it is needed here to decrypt the data. The sender ID is on a sticker
+in the battery compartment. There are three groups of three digits there. The
+last six digits are your sender ID. Eg "400 617 633" gives you the sender id 
+617633. This number goes into IKEA_SPARSNAS_SENSOR_ID in this file.
+
+
+The data is sent using GFSK modulation. It requires PD_MIN_PULSE_SAMPLES in 
+pulse_detect.h to be lowered to 5 to be able to demodulate at 250kS/s. The
+preamble is optimally 4 bytes of 0XAA. Then the sync word 0xD201. Here only 
+the last 2 bytes of the 0xAA preamble is checked, as the first ones seems
+to be corrupted quite often. There are plenty of integrety checks made on 
+the demodulated package which makes this compromise OK.
+
+Packet structure according to: https://github.com/strigeus/sparsnas_decoder 
+(with some changes by myself)
+
+0:  uint8_t length;        // Always 0x11
+1:  uint8_t sender_id_lo;  // Lowest byte of sender ID
+2:  uint8_t unknown;       // Not sure
+3:  uint8_t major_version; // Always 0x07 - the major version number of the sender.
+4:  uint8_t minor_version; // Always 0x0E - the minor version number of the sender.
+5:  uint32_t sender_id;    // ID of sender
+9:  uint16_t sequence;     // Sequence number of current packet
+11: uint16_t effect;       // Current effect usage
+13: uint32_t pulses;       // Total number of pulses
+17: uint8_t battery;       // Battery level, 0-100%
+18: uint16_t CRC;          // 16 bit CRC of bytes 0-17
+
+The packet's integrety can be checked with the 16b CRC at the end of the packet. 
+There are also several other ways to check the integrety of the package. 
+ - (preamble)
+ - CRC
+ - The decrypted sensor ID
+ - the constant bytes at 0, 3 and 4
+
+The decryption, CRC is calculation, value extraction and interpretation is 
+taken from https://github.com/strigeus/sparsnas_decoder and adapted to 
+this application. Many thanks to strigeus!
+
+Most other things are from https://github.com/kodarn/Sparsnas which is an
+amazing repository of the IKEA Sparsnäs. Everything is studied with greay
+detail. Many thanks to kodarn!
+
+*/
+
+
+
+#define IKEA_SPARSNAS_SENSOR_ID 617633
+#define IKEA_SPARSNAS_PULSES_PER_KWH 1000
+
```

Comment:
We need to figure out how to pass along decoder parameters. This needs to be in a config file.

Reply:
Perhaps we can think up a clean way to unify this, the recent WH5 quirk, and flex specs. Something like parametric decoders.

Reply:
I already snuck that in when refactoring the decoders ;)
There is a private context (flex currently, but good for all parametric decoders): https://github.com/merbanan/rtl_433/blob/master/include/rtl_433_devices.h#L151
Usage example is https://github.com/merbanan/rtl_433/blob/master/src/devices/flex.c#L104
We now "just" need a way to declare and instantiate parametric decoders in a general and enumerable way.